### DISCUSSION TITLE
The new ASR rules should be set to Audit or Warn state until they graduate to stable.

### DISCUSSION THREAD

**Discussion by agpt8** [HotCakeX/Harden-Windows-Security#219] [General]
The new ASR rules added recently to the [ASR table](https://learn.microsoft.com/en-us/microsoft-365/security/defender-endpoint/attack-surface-reduction-rules-reference?view=o365-worldwide#asr-rule-to-guid-matrix) and recently enforced by the script/module in the PR #212, should be set to Audit (status ID 2) or Warn (status ID 6) until they graduate to stable.

The table shows the new rules that are in preview. And for sure, they are a little too restrictive, specifically Block use of copied or impersonated system tools (preview) (ID: c0033c00-d16d-4114-a5a0-dc9b3a7d2ceb).

So far, this particular rule has been the most problematic for me. It has blocked several background git operations when working with several repositories, java specific executables from jetbrains IDEs, vscode and zulu (klist.exe, 7z.exe, and more), NVIDIA driver updates as it tried to extract to temp folder and tried running rundll32.exe as part of the upgrade, standalone 7z operations and many more.

As far as I can tell, many of these executables like 7z and klist are definitely not part of the system tools and it still blocked them.

I was able to mitigate this by changing the value for the rule ID in GPO to 6.

IMO, they are too unstable and way too restrictive for now and should be set to audit or warn state until they are stable enough.

I would like to hear your thoughts on this. Have you had any issues on your side?

**Comment by HotCakeX** [comment]
Hi,
Do you see logs for ASRs in here

<img width="269" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/d491c189-6cb9-47dc-beb6-94a4d0f9e18f">

<br>

I really like to look at them. I haven't had any issues with the new ASR rules. I use VS code, GitHub desktop that uses Git executables under the hood, i use Nanazip that uses 7zip executables under the hood, and I have Nvidia driver installed (latest game ready version).

**Reply by HotCakeX** [reply]
Correction, i don't have latest driver installed, I have 551.76 but i see 551.86 is now available (released 3 days ago), downloading and upgrading, will let you know how it goes.

btw I have everything enabled on my system including smart app control.

**Reply by HotCakeX** [reply]
By the way you're on Insider beta channel like me, right? OS build 22635.3350

**Reply by agpt8** [reply]
Yup! I am on the latest beta.

~Let me figure out how to export the required logs 😅. I'll attach them in a few minutes~

Update: Here are the logs
There are two saved views. One containing warning only, the other one containing all the events for ASR. Surprisingly, there are a lot of Information level events.
[ASR events.zip](https://github.com/HotCakeX/Harden-Windows-Security/files/14724935/ASR.events.zip)

**Reply by HotCakeX** [reply]
On the sidebar in Event Viewer, after selecting the Attack Surface Reduction events you can select this 🙂

You can email them to me too if you don't wanna publicly put it here

<img width="238" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/c5f30916-12c5-4ef3-a24f-2af46b9a09fb">

<br>

<br>

P.S now Nvidia driver upgraded

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/ee646b4e-9f85-4c26-aa23-46bb8d66f232)

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/3a717a63-90b7-4c59-870d-3519190d4999)

**Reply by agpt8** [reply]
I used the new NVIDIA app to install the latest driver. The app does not show these windows, and everything happens in the custom window it draws (tbh these windows that the driver draws are quite old, and the app is now drawing its own windows while updating).

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/16762897/66057c9f-8732-4244-9baa-1e86b516356d)
![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/16762897/bcb58f47-6b5d-4a97-9fe3-54e2644535e4)

Now that I have changed the ID for that rule, it no longer errors out. I didnt test it with the old experience.

**Reply by HotCakeX** [reply]
Ah I have to try out the new app!

But you're right that new ASR rule is causing problem for your programs, it's obvious in the logs, i think the best thing to do is setting it as an optional sub-category for the ASR category so it doesn't get applied with the rest of the rules, at least until it exits the preview phase.

Sorry about that, i'll change it asap

**Reply by agpt8** [reply]
Oh thank you! Thats great to hear! I'll mark this as complete 😃

**Reply by HotCakeX** [reply]
🙏🙏
